source/common/grpc/context_impl.cc

```diff
@@ -103,7 +103,12 @@ ContextImpl::resolveServiceAndMethod(const Http::HeaderEntry* path) {
   if (path == nullptr) {
     return request_names;
   }
-  const auto parts = StringUtil::splitToken(path->value().getStringView(), "/");
+  absl::string_view str = path->value().getStringView();
+  size_t query_offset = str.find('?');
+  if (query_offset != absl::string_view::npos) {
```

Comment:
You don't need this if, 2nd parameter of `substr` is default to `npos`.

Reply:
Removed if block.